### PR TITLE
jcroteau/APPEALS-28925 - Update to Rails `6.1.Z`

### PR DESCRIPTION
**Pull request by jcroteau**
Resolves https://jira.devops.va.gov/browse/APPEALS-28925

# Description

## Upgrade Steps
- Updated rails and other dependencies as necessary in `Gemfile`: 
    - `gem "rails", "6.1.7.6"`
    - `gem "activerecord-oracle_enhanced-adapter", "~> 6.1.0"`
    - `gem "caseflow", "~> 1.0.0", git: "https://github.com/department-of-veterans-affairs/caseflow-commons"`
    - `gem "tzinfo", "~> 2.0"`
    - `gem "bullet", "~> 6.1.0"` (for ActiveRecord 6.1 support)
    - `gem "activerecord-import", "1.0.3"` (for ActiveRecord 6.1 support)
    - `gem "redis", "~> 4.0.1"` (constrain version to prevent breaking updates)
- Updated `Gemfile.lock`
    - `bundle update rails activerecord-oracle_enhanced-adapter caseflow tzinfo bullet activerecord-import`
- Generated configuration changes and `new_framework_defaults_6_1.rb`:
    - `bin/rails app:update`
    - **Note:** You can reference [RailsDiff](https://railsdiff.org/6.0.6.1/6.1.7.6) to preview configuration changes between versions.
- Removed `bin/rails app:update`-generated DB migrations for ActiveStorage, since we are not using ActiveStorage.
- Inspected `bin/rails app:update`-generated config changes and decided what to keep and what to replace.
- Moved `new_framework_defaults_6_1.rb` settings into `config/application.rb` and overrided those where necessary.

## Testing Plan
- `#TODO Add link to Regression Test Plan in Jira`

- [ ] For feature branches merging into master: Was this deployed to UAT?

# Backend
# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [ ] No new code climate issues added

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ gem "bgs", git: "https://github.com/department-of-veterans-affairs/ruby-bgs.git"
 gem "bootsnap", require: false
 gem "browser"
 gem "business_time", "~> 0.9.3"
-gem "caseflow", git: "https://github.com/department-of-veterans-affairs/caseflow-commons", ref: "6377b46c2639248574673adc6a708d2568c6958c"
+gem "caseflow", git: "https://github.com/department-of-veterans-affairs/caseflow-commons", branch: "AlecK/APPEALS-27013"
```
Comment (jcroteau) [inline Gemfile:20]: 🎗️ **Reminder:** After `caseflow-commons 1.0.0` is merged, update this line and `bundle update caseflow`:

```suggestion
gem "caseflow", "~> 1.0.0", git: "https://github.com/department-of-veterans-affairs/caseflow-commons"
```

File: Gemfile.lock [hunk 2]
```diff
@@ -9,16 +9,14 @@ GIT
 
 GIT
   remote: https://github.com/department-of-veterans-affairs/caseflow-commons
-  revision: 6377b46c2639248574673adc6a708d2568c6958c
-  ref: 6377b46c2639248574673adc6a708d2568c6958c
+  revision: edd91cc2d3bcbdb84f7637bdfe5bcfa20c583759
+  branch: AlecK/APPEALS-27013
   specs:
-    caseflow (0.4.8)
+    caseflow (1.0.0)
```
Comment (jcroteau) [inline Gemfile.lock:15]: See related [**Reminder** comment](https://github.com/department-of-veterans-affairs/caseflow/pull/19760/files#r1366236670) above.

File: Gemfile.lock [hunk 3]
```diff
@@ -157,33 +159,33 @@ GEM
...
+    aws-sdk-core (2.11.632)
       aws-sigv4 (~> 1.0)
       jmespath (~> 1.0)
-    aws-sdk-resources (2.10.112)
-      aws-sdk-core (= 2.10.112)
-    aws-sigv4 (1.0.2)
+    aws-sdk-resources (2.11.632)
+      aws-sdk-core (= 2.11.632)
+    aws-sigv4 (1.6.0)
+      aws-eventstream (~> 1, >= 1.0.2)
```
Comment (jcroteau) [inline Gemfile.lock:171]: 📝 **Note to self:** The AWS gem updates were pulled in by the `caseflow` gem dependency on `aws-sdk`. We can (and should) likely revert these.

File: Gemfile.lock [hunk 4]
```diff
@@ -222,7 +224,7 @@ GEM
     crack (0.4.3)
       safe_yaml (~> 1.0.0)
     crass (1.0.6)
-    d3-rails (7.0.0)
+    d3-rails (7.8.5)
```
Comment (jcroteau) [inline Gemfile.lock:227]: 📝 **Note to self:** The `d3-rails` update was pulled in by the `caseflow` gem dependency on `d3-rails`. We can (and should) likely revert this.

File: Gemfile.lock
Comment (jcroteau) [inline Gemfile.lock:1]: 📝 **Note to self:** I'll likely need to re-do the `Gemfile.lock` changes with a more conservative `bundle update`, as it appears there are some unnecessary gem updates in here...

**Comment by ThorntonMatthew** [comment]
Closing due to inactivity as part of our post-PI cleanup effort. Feel free to reopen if this PR is still desired.

**Pull request closed without merging**